In [8]:
import spacy
nlp = spacy.load('en_core_web_md')
word1 = nlp("cat")
word2 = nlp("monkey")
word3 = nlp("banana")
word4 = nlp("dog")
print(word1.similarity(word2))
print(word3.similarity(word2))
print(word3.similarity(word1))
print(word1.similarity(word4))

0.5929930274321619
0.40415016164997786
0.22358825939615987
0.8220816752553904


# **Insights**

* The similarity score between "cat" and "monkey" is higher than expected, indicating that these two words share some semantic properties. This could be because they are both animals, or because they have some physical or behavioral similarities.

* The similarity score between "banana" and "monkey" is lower than expected, which suggests that these two words have less in common in terms of their semantic properties.

* The similarity score between "cat" and "dog" is relatively high, which is not surprising given that these two words are both common household pets and share some physical and behavioral characteristics.



In [7]:
sentence_to_compare = "Why is my cat on the car"
sentences = ["where did my dog go",
"Hello, there is my car",
"I\'ve lost my car in my car",
"I\'d like my boat back",
"I will name my dog Diana"]
model_sentence = nlp(sentence_to_compare)
for sentence in sentences:
    similarity = nlp(sentence).similarity(model_sentence)
    print(sentence + " - ", similarity)

where did my dog go -  0.630065230699739
Hello, there is my car -  0.8033180111627156
I've lost my car in my car -  0.6787541571030323
I'd like my boat back -  0.5624940517078084
I will name my dog Diana -  0.6491444739190607


# **Insights**

The sentence "Hello, there is my car" has the highest similarity score (0.803) with the input sentence, which is expected since both sentences mention the word "car" and have a similar structure.

The sentence "I'd like my boat back" has the lowest similarity score (0.562) with the input sentence, which is also expected since the two sentences have very little in common in terms of their content or structure.

The other sentences have intermediate similarity scores ranging from 0.630 to 0.679, which suggests that they have some shared semantic properties with the input sentence, but also some differences.

# **Difference between en_core_web_sm and en_core_web_md Model**

The main difference between the 'en_core_web_sm' and 'en_core_web_md' models in SpaCy is their size and the amount of training data used to create them. The 'en_core_web_sm' model is a small model that uses a smaller number of embeddings and fewer parameters, making it faster and more lightweight than the larger 'en_core_web_md' model.

* If speed and efficiency are important, the 'en_core_web_sm' model might be a good choice. 
* If accuracy and the ability to handle more complex language structures are essential, the 'en_core_web_md' model might be a better option.